src/Psalm/Internal/Analyzer/Statements/Block/ForeachAnalyzer.php

```diff
@@ -537,8 +537,7 @@ public static function checkIteratorType(
                         continue;
                     }
 
-                    $value_type_part = $iat->type_params[1];
-                    $key_type_part = $iat->type_params[0];
+                    [$value_type_part, $key_type_part] = $iat->type_params;
```

Comment:
```suggestion
                    [$key_type_part, $value_type_part] = $iat->type_params;
```
Vars were in the reverse order in the original code.

Reply:
well spotted! I was still trying to see what could have caused the issue in CI! Thanks!